internal/daemon/worker/controller_connection.go

```diff
@@ -442,5 +442,7 @@ func monitorUpstreamConnectionState(ctx context.Context, cc *grpc.ClientConn, co
 		}
 
 		connectionState.Store(newState)
+
+		time.Sleep(10 * time.Millisecond)
```

Comment:
Is this intended to be a long term solution or just a hotfix that we can get out quickly to users? It feels like we need to spend more time digging into the source of the issue. Why is the state changing so often? It could indicate something about the application that's worrying.

Reply:
Pushed another fix which should address the issue. Took out sleep change